src/actix/api/cluster_api.rs

```diff
@@ -14,19 +21,33 @@ async fn cluster_status(dispatcher: web::Data<Dispatcher>) -> impl Responder {
 }
 
 #[delete("/cluster/peer/{peer_id}")]
-async fn remove_peer(dispatcher: web::Data<Dispatcher>, peer_id: web::Path<u64>) -> impl Responder {
+async fn remove_peer(
+    dispatcher: web::Data<Dispatcher>,
+    peer_id: web::Path<u64>,
+    web::Query(force): web::Query<Force>,
+) -> impl Responder {
     let timing = Instant::now();
     let dispatcher = dispatcher.into_inner();
     let peer_id = peer_id.into_inner();
 
-    if dispatcher.peer_has_shards(peer_id).await {
+    let (has_shards, has_the_only_shard) = dispatcher.peer_has_shards(peer_id).await;
```

Comment:
Is this guarding validation protecting us from the case where two remove_peer calls are made in parallel?

Reply:
This is for handling the case where there is only 1 shard in collection - and therefore we cannot delete this shard. I added a comment below